src/WorkItemMigrator/tests/Migration.Jira-Export.Tests/JiraItemTests.cs

```diff
@@ -24,6 +28,126 @@ public void Setup()
             _fixture.Customize(new AutoNSubstituteCustomization() { });
         }
 
+        [Test]
+        public void When_an_attachment_is_added_Then_it_will_be_migrated()
+        {
+            //Arrange
+            var provider = _fixture.Freeze<IJiraProvider>();
+            long issueId = _fixture.Create<long>();
+            string issueKey = _fixture.Create<string>();
+            string attachmentId = _fixture.Create<int>().ToString();
+            string attachmentName = _fixture.Create<string>();
+
+            var fields = JObject.Parse($@"{{
+                'issuetype': {{ 'name': 'Story' }},
+                'attachment': [
+                {{
+                  'self': 'https://server/rest/api/2/attachment/{attachmentId}',
+                  'id': '{attachmentId}',
+                  'filename': '{attachmentName}',
+                  'author': null,
+                  'created': '{DateTime.Now.ToString("yyyy - MM - ddTHH:mm: ss.fffZ")}',
+                  'size': '{_fixture.Create<int>()}',
+                  'mimeType': 'image/jpeg',
+                  'content': 'https://server/rest/api/2/attachment/{HttpUtility.UrlEncode(attachmentName)}',
+                }}
+              ]
+            }}");
+            var renderedFields = JObject.Parse("{ 'custom_field_name': 'SomeValue', 'description': 'RenderedDescription' }");
+
+            var changelog = new List<JObject>() {
+                new HistoryItem() // add attachment
+                {
+                    Field = "Attachment",
+                    FieldType = "jira",
+                    From = null,
+                    FromString = null,
+                    To = attachmentId,
+                    ToString = attachmentName
+                }.ToJObject()
+            };
+
+            JObject remoteIssue = new JObject
+            {
+                { "id", issueId },
+                { "key", issueKey },
+                { "fields", fields },
+                { "renderedFields", renderedFields }
+            };
+
+            provider.DownloadIssue(default).ReturnsForAnyArgs(remoteIssue);
+            provider.DownloadChangelog(default).ReturnsForAnyArgs(changelog);
+            var jiraSettings = createJiraSettings();
+            provider.GetSettings().ReturnsForAnyArgs(jiraSettings);
+
+            //Act
+            var jiraItem = JiraItem.CreateFromRest(issueKey, provider);
+
+            //Assert
```

Comment:
Again, when using multiple asserts it is good practice to wrap it with Assert.Multiple, in that case it will run all asserts and not fail on the first on

Reply:
Resolved